### PR TITLE
Add some configuration checks to highlight probably production issues

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Comment by snazy** [comment]
Emits warnings like this on startup:
```
2024-02-08 10:05:24,463 WARN  [org.pro.qua.pro.sto.PersistProvider] (main) IN_MEMORY version-store warning:

****************************************************************************************************
** The configured version store type is IN_MEMORY is only for testing purposes and experimentation.
** IN_MEMORY is NOT for production use!
** ALL DATA WILL BE LOST if the process is shut down, nothing is persisted.
**
** Recommended action:
** --> Use a supported database, see https://projectnessie.org/try/configuration/
****************************************************************************************************

2024-02-08 10:05:24,463 WARN  [org.pro.qua.pro.sto.PersistProvider] (main) No authorization & no authorization warning:

****************************************************************************************************
** Both authentication (AuthN) and authorization (AuthZ) are disabled,
** all requests to Nessie will be permitted.
**
** This means: EVERYBODY with access to Nessie can read, write and change everything.
**
** Recommended action:
** --> Enable AuthN & AuthZ, see https://projectnessie.org/try/configuration/
****************************************************************************************************

2024-02-08 10:05:24,496 INFO  [io.quarkus] (main) nessie-quarkus 0.76.7-SNAPSHOT on JVM (powered by Quarkus 3.7.1) started in 1.819s. Listening on: http://0.0.0.0:19120
```

**Comment by adutra** [comment]
I'm not a fan of these multiline warnings surrounded by asterisks tbh. For a few reasons:

- They are 100 chars wide, could end up garbled in smaller terminals.
- Messages could end up garbled/split in log collectors.
- For people trying out Nessie for the first time by simply running a docker container, they could be too scary. We just got rid of 2 annoyances (OIDC stack traces and SEVERE log messages when OIDC or telemetry is not configured), which imho improve the UX for people new to Nessie. These warnings are going to degrade the newcomers UX again imho.

How about just one or two lines of warnings? E.g.

```
2024-02-08 10:05:24,463 WARN  [org.pro.qua.pro.sto.PersistProvider] (main) The configured version store type IN_MEMORY is only for testing purposes and experimentation. In production, please use a supported database, see https://projectnessie.org/try/configuration/ for details.
```
or
```
2024-02-08 10:05:24,463 WARN  [org.pro.qua.pro.sto.PersistProvider] (main) The configured version store type IN_MEMORY is only for testing purposes and experimentation.
2024-02-08 10:05:24,463 WARN  [org.pro.qua.pro.sto.PersistProvider] (main) In production, please use a supported database, see https://projectnessie.org/try/configuration/ for details.
```

File: servers/quarkus-server/src/main/java/org/projectnessie/server/configchecks/ConfigChecks.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  // This function shall be called on startup. The actually returned object is currently
+  // meaningless.
+  @Produces
+  @Singleton
+  @Startup
+  public ConfigCheck configCheck() {
+    if (versionStoreConfig.getVersionStoreType() == VersionStoreConfig.VersionStoreType.IN_MEMORY) {
+      LOGGER.warn(
+          "Non-production IN_MEMORY version-store warning: "
```
Comment (dimas-b) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/configchecks/ConfigChecks.java:46]: nit: do we really need this message prefix (similarly the prefixes in the following messages)?

**Review by dimas-b — COMMENTED** [review]
LGTM overall.

**Review by dimas-b — APPROVED** [review]

**Pull request merged**